bluemira/equilibria/fem_fixed_boundary/utilities.py

```diff
@@ -59,30 +68,345 @@ class ScalarSubFunc(dolfin.UserExpression):
+            "or alist of them."
+        )
 
     def eval_cell(self, values, x, cell):
         """Evaluate the value on each cell"""
-        m = self.subdomains[cell.index]
-        if m in self.markers:
-            index = np.where(np.array(self.markers) == m)
-            func = self.functions[index[0][0]]
-            if callable(func):
-                values[0] = func(x)
-            elif isinstance(func, (int, float)):
-                values[0] = func
+        if self.markers is None or self.subdomains is None:
+            func = self.functions[0]
+        else:
+            m = self.subdomains[cell.index]
+            if m in self.markers:
+                index = np.where(np.array(self.markers) == m)
+                func = self.functions[index[0][0]]
             else:
-                raise ValueError(f"{func} is not callable or is not a constant")
+                func = 0
+        if callable(func):
+            values[0] = func(x)
+        elif isinstance(func, (int, float)):
+            values[0] = func
         else:
-            values[0] = 0
+            raise ValueError(f"{func} is not callable or is not a constant")
 
     def value_shape(self):
         """
         Value_shape function (necessary for a UserExpression)
         https://fenicsproject.discourse.group/t/problems-interpolating-a-userexpression-and-plotting-it/1303
         """
         return ()
+
+
+def plot_scalar_field(
+    x, y, data, levels=20, axis=None, contour=True, tofill=True, **kwargs
+):
+    """
+    Plot a scalar field
+
+    Parameters
+    ----------
+    x: np.array(n, m)
+        x coordinate array
+    z: np.array(n, m)
+        z coordinate array
+    data: np.array(n, m)
+        value array
+    levels: int
+        Number of contour levels to plot
+    axis: Optional[Axis]
+        axis onto which to plot
+    contour: bool
+        Whether or not to plot contour lines
+    tofill: bool
+        Whether or not to plot filled contours
+
+    Returns
+    -------
+    axis: Axis
+        Matplotlib axis on which the plot ocurred
+    """
+    if axis is None:
+        fig = plt.figure()
+        axis = fig.add_subplot()
+    else:
+        fig = plt.gcf()
+
+    defaults = {"linewidths": 2, "colors": "k"}
+    contour_kwargs = {**defaults, **kwargs}
+
+    cntr = None
+    cntrf = None
+
+    if contour:
+        cntr = axis.tricontour(x, y, data, levels=levels, **contour_kwargs)
+
+    if tofill:
+        cntrf = axis.tricontourf(x, y, data, levels=levels, cmap="RdBu_r")
+        fig.colorbar(cntrf, ax=axis)
+
+    axis.set_xlabel("x [m]")
+    axis.set_ylabel("z [m]")
+    axis.set_aspect("equal")
+
+    return axis, cntr, cntrf
+
+
+def plot_profile(x, prof, var_name, var_unit):
+    """
+    Plot profile
+    """
+    fig, ax = plt.subplots()
+    ax.plot(x, prof)
+    ax.set(xlabel="x (-)", ylabel=var_name + " (" + var_unit + ")")
+    ax.grid()
+    plt.show()
```

Comment:
We may want to consider making this `show` optional, as we usually do for plotting.